ui/src/EnvVarForm/components/EnvVarForm.js

```diff
@@ -0,0 +1,156 @@
+            var key = keyValuePair.children[0].children[0]
+            var value = keyValuePair.children[1]
+            if (key.value) {
+                if (value.value) {
+                    vars[key.value] = value.value
+                }
+                else {
+                    vars[key.value] = value.placeholder
+                }
+            }
+            else {
+                if (value.value) {
+                    vars[key.placeholder] = value.value
+                }
+                else {
+                    vars[key.placeholder] = value.placeholder
+                }
+            }
+        }
+        this.setState({envvars: vars}, function() {this.sendVars()})
+    }
+
+    removeForm(e) {
+        var parent = e.target.parentElement
+        parent.remove()
+        this.onBlur()
+    }
+
+    enterVariable(event) {
+        var dropdown = event.target.parentElement.parentElement
+        var key = dropdown.previousSibling
+        key.placeholder = event.target.text
+    }
+
+    addForm(e){
+        //Frequently used variables could also be entered as a list if there are too many of them.
+        const newItem = [
+            <div className="EnvVarGrid">
+                <FormControl name='key' type="text" placeholder='key' onBlur={(e) => this.onBlur(e)}/>
+                <DropBtn id="EnvVarDrop" innerClass="EnvVarDropdown" text=''>
+                    <Dropdown.Item onClick={(e) => this.enterVariable(e)} eventKey="1">AWS_ACCESS_KEY_ID</Dropdown.Item>
+                    <Dropdown.Item onClick={(e) => this.enterVariable(e)} eventKey="2">AWS_SECRET_ACCESS_KEY</Dropdown.Item>
+                </DropBtn>
+            </div>,
+            <FormControl className="InnerGap" type="text" placeholder='value' onBlur={(e) => this.onBlur(e)}/>,
+            <Button className="InnerGap" variant='danger' onClick={(e) => this.removeForm(e)}>
+                Remove
+            </Button>
+            ]
+        this.setState(previousState => ({
+            items: [...previousState.items, newItem]
+        }));
+    }
+
+    renderForms() {
+        console.log('Logging envvars:', this.state.envvars)
+        for (const [key, value] of Object.entries(this.state.envvars)) {
+            const newItem = [
+                    <div className="EnvVarGrid">
+                        <FormControl name={key} type="text" placeholder={key} onBlur={(e) => this.onBlur(e)}/>    
+                        <DropBtn id="EnvVarDrop" innerClass="EnvVarDropdown" text=''>
+                            <Dropdown.Item onClick={(e) => this.enterVariable(e)} eventKey="1">AWS_ACCESS_KEY_ID</Dropdown.Item>
+                            <Dropdown.Item onClick={(e) => this.enterVariable(e)} eventKey="2">AWS_SECRET_ACCESS_KEY</Dropdown.Item>
+                        </DropBtn>
+                    </div>,
+                    <FormControl className="InnerGap" type="text" placeholder={value} onBlur={(e) => this.onBlur(e)}/>,
```

Comment:
Can we make this `type="password"` if the env name is one of the AWS credentials?

For the future, we will want to refactor the env var definition to also hold type (text/password) to make sure users can add secrets "securely"